### PR TITLE
Fixed singleton issue when leaving several nodes

### PR DESCRIPTION
**Pull request by ismaelhamed**
Issue [#27487](https://github.com/akka/akka/issues/27487)
Port [#27488](https://github.com/akka/akka/pull/27488)

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Using this to help guinea pig the new TRX reporter for the MNTR

File: src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs [hunk 1]
```diff
@@ -632,13 +634,11 @@ private ActorSelection Peer(Address at)
...
-        private void GetNextOldestChange()
+        private void GetNextOldestChanged()
         {
-            if (_oldestChangedReceived)
-            {
-                _oldestChangedReceived = false;
-                _oldestChangedBuffer.Tell(OldestChangedBuffer.GetNext.Instance);
-            }
+            if (!_oldestChangedReceived) return;
+            _oldestChangedReceived = false;
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs]: So if I understand this correctly, this code can never run inside the `ClusterSingletonState.Start`, right?

Comment (ismaelhamed) [reply src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs]: This code hasn't change, I just inverted the "if" statement.

Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs]: Got it

**Pull request merged**